src/GafferSceneUI/StandardLightVisualiser.cpp

```diff
@@ -237,74 +237,75 @@ IECoreGL::ConstRenderablePtr StandardLightVisualiser::visualise( const IECore::I
 	const Color3f finalColor = color * intensity * pow( 2.0f, exposure );
 
 	GroupPtr result = new Group;
+	GroupPtr ornaments = new Group;  // Ornaments are affected by visualiser:scale while
+	GroupPtr geometry = new Group;   // geometry isn't as its size matters for rendering.
+	result->addChild( ornaments );
+	result->addChild( geometry );
 
-	/// \todo This is problematic for a few reasons :
-	///
-	/// - We don't actually want to apply it to the area light shapes created below for "quad" etc.
-	/// - We should find a better way to opt out of expensive visualisations (in particular for
-	///   large environment light textures)
-	///
-	/// Since this feature is only used by lights internal to Image Engine, we can ignore all this for
-	/// now, but it would be good to address in the future.
 	const FloatData *visualiserScaleData = attributes->member<FloatData>( "visualiser:scale" );
 	float visualiserScale = visualiserScaleData ? visualiserScaleData->readable() : 1.0;
 
+	/// \todo: We should find a better way to opt out of expensive visualisations
+	///        (in particular for large environment light textures)
 	if( visualiserScale == 0 )
 	{
 		return result;
 	}
 
-	Imath::M44f topTrans;
+	Imath::M44f topTransform;
 	if( orientation )
 	{
-		topTrans = orientation->readable();
+		topTransform = orientation->readable();
 	}
-	topTrans.scale( V3f( visualiserScale ) );
-	result->setTransform( topTrans );
+	result->setTransform( topTransform );
+
+	Imath::M44f ornamentsTransform;
```

Comment:
As a post-merge aside, I used to find this "make the matrix, scale the matrix" cumbersome until I realised that you can do `M44f().scale()` to achieve the same thing. The nice thing about that is you can get better const correctness : `const M44f ornamentsTransform = M44f().scale( V3f( visualiserScale ) )`. Or you can omit the variable entirely : `ornaments->setTransform( M44f().scale( V3f( visualiserScale ) ) )`.

Reply:
Ah, that's much better, indeed. Thank you :)